packages/react-events/src/__tests__/Swipe-test.internal.js

```diff
@@ -0,0 +1,326 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @emails react-core
+ */
+
+'use strict';
+
+let React;
+let ReactFeatureFlags;
+let ReactDOM;
+let Swipe;
+
+const createMouseEvent = (type, x = 0, y = 0) => {
+  const event = document.createEvent('MouseEvents');
+  event.initMouseEvent(type, true, true, window, 1, x, y, 0, 0);
+  return event;
+};
+
+const createTouchEvent = (type, x = 0, y = 0, identifiers = [1]) => {
+  let touches = [];
+  for (let i = 0; i < identifiers.length; i++) {
+    touches.push({
+      identifier: identifiers[i],
+      target: document,
+      clientX: 0,
+      clientY: 0,
+      screenX: x,
+      screenY: y,
+    });
+  }
+
+  const event = new TouchEvent(type, {
+    cancelable: true,
+    bubbles: true,
+    composed: true,
+    touches,
+    targetTouches: touches,
+    changedTouches: touches,
+  });
+  return event;
+};
+
+describe('Swipe event responder', () => {
+  let container;
+
+  beforeEach(() => {
+    jest.resetModules();
+    ReactFeatureFlags = require('shared/ReactFeatureFlags');
+    ReactFeatureFlags.enableEventAPI = true;
+    React = require('react');
+    ReactDOM = require('react-dom');
+    Swipe = require('react-events/swipe');
+
+    container = document.createElement('div');
+    document.body.appendChild(container);
+  });
+
+  afterEach(() => {
+    document.body.removeChild(container);
+    container = null;
+  });
+
+  describe('onSwipeStart', () => {
+    let onSwipeStart, ref;
+
+    beforeEach(() => {
+      onSwipeStart = jest.fn();
+      ref = React.createRef();
+      const element = (
+        <Swipe onSwipeStart={onSwipeStart}>
+          <div ref={ref} />
+        </Swipe>
+      );
+      ReactDOM.render(element, container);
+    });
+
+    it('is called after "pointerdown" event', () => {
+      ref.current.dispatchEvent(createMouseEvent('pointerdown'));
```

Comment:
This is dispatching a `MouseEvent`

Reply:
based on https://developer.mozilla.org/en-US/docs/Web/API/Pointer_events pointer event types are intentionally similar to mouse event types, however, PointerEvent API is not defined and document.createEvent('Event') deprecated and does not have screenX/Y option too. so, until finding a way to use Pointer Event API, I'm using MouseEvent.